Runtime/JSSynchronizationContext.cs

```diff
@@ -0,0 +1,68 @@
+using System;
+using System.Threading;
+
+namespace NodeApi;
+
+public class JSSynchronizationContext : SynchronizationContext, IDisposable
+{
+    private readonly JSThreadSafeFunction _tsfn;
+    private readonly SynchronizationContext? _previousSyncContext;
+
+    public bool IsDisposed { get; private set; }
+
+    public JSSynchronizationContext()
+    {
+        _tsfn = JSThreadSafeFunction.New(
+            maxQueueSize: 0,
+            initialThreadCount: 1,
+            asyncResourceName: (JSValue)"SynchronizationContext");
+
+        _previousSyncContext = Current;
+        SetSynchronizationContext(this);
+    }
+
+    ~JSSynchronizationContext()
+    {
+        Dispose(disposing: false);
+    }
+
+    public void Dispose()
+    {
+        Dispose(disposing: true);
+        GC.SuppressFinalize(this);
+    }
+
+    public override void Post(SendOrPostCallback callback, object? state)
+    {
+        if (!IsDisposed)
+        {
+            _tsfn.NonBlockingCall(() => callback(state));
+        }
+    }
+
+    public override void Send(SendOrPostCallback callback, object? state)
+    {
+        if (this == Current)
+        {
+            callback(state);
+        }
+        else
+        {
+            Post(callback, state);
```

Comment:
I checked a few other implementations of `SynchronizationContext.Send()`, and they all synchronously wait for the callback execution to complete. But this looks like it just sends the message without waiting. One way to fix that could be to wrap the callback in a lambda that sets a `ManualResetEvent` after the inner callback returned, then wait on that callback in the outer method. Or maybe you have another idea.

Or if `Send()` is not used anywhere, maybe this doesn't matter.

Reply:
Good point! Changed to have a true synchronous call. I often use this `InvokeElsePost` pattern and did it here because I thought docs say it too. Re-reading it shows that they were not saying it. It was just a wishful thinking on my side. :)